domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2611,20 +2613,17 @@ public ResultsHelper(SQLRowStream sqlRowStream, Class<T> clazz) {
    * @param o
    * @param row
    */
-  void populateExternalColumns(Map<String, Method> externalColumnSetters, Object o, JsonObject row) {
+  void populateExternalColumns(Map<String, Method> externalColumnSetters, Object o, Row row)
+      throws InvocationTargetException, IllegalAccessException {
     for (Map.Entry<String, Method> entry : externalColumnSetters.entrySet()) {
       String columnName = entry.getKey();
       Method method = entry.getValue();
-      try {
-        Object value = row.getValue(columnName);
-        if (value instanceof JsonArray) {
-           method.invoke(o, Json.decodeValue(((JsonArray) value).encode(), method.getParameterTypes()[0]));
-        } else {
-          method.invoke(o, value);
+        String[] stringArray = row.getStringArray(columnName);
+        if (stringArray != null) {
+          method.invoke(o, Arrays.asList(stringArray));
+          continue;
```

Comment:
Using if-else is more appropriate and intuitive because the if block and the else block do the same thing, they differ only in the 2nd argument.

Reply:
Oh yes. This was cut'n paste from old code.